core-bundle/src/Resources/contao/library/Contao/Automator.php

```diff
@@ -471,6 +471,12 @@ public function rotateLogs()
 
 		foreach ($arrFiles as $strFile)
 		{
+			// Ignore Monolog log files (see #2579)
+			if (preg_match('/-\d{4}-\d{2}-\d{2}\.log$/', $strFile))
```

Comment:
```suggestion
			if (preg_match('/(-|_)\d{1,4}(-|_|\.)\d{1,2}(-|_|\.)\d{1,4}\.log$/', $strFile))
```

This would also match
 - `foo_2020-01-01.log`
 - `foo-2020.01.01.log`
 - `foo-01.01.2020.log`
 - `foo__2020-1-1.log`
 - `foo__1_1_2020.log`

Reply:
Why? We never use these formats, do we?

Reply:
This isn't about "we" the Contao core, because we don't use the old log handler anymore? `log_message` does not create a file with a date, so such files would "manually" be created by developers use the old method, and they're free to choose the date format?

Reply:
I think you are missing the point of the if clause. It does not define which files to **handle** but which files to **ignore**. And in my opinion, we must only ignore the Monolog date format, which is:

* RotatingFileHandler::FILE_PER_DAY ("Y-m-d")
* RotatingFileHandler::FILE_PER_MONTH ("Y-m")
* RotatingFileHandler::FILE_PER_YEAR ("Y")

https://github.com/Seldaek/monolog/blob/main/src/Monolog/Handler/RotatingFileHandler.php#L87-L90

So if a log file is named `foo-01.01.2020.log`, we can assume that it is **not** handled by Monolog and that we have to rotate it ourselves.

Reply:
My initial thought was to not rotate files that already include a date, because they are "recreated" each day anyway. However, that would not **delete** those files after 10 days, which is what the Automator is supposed to do. Therefore I agree we should only match on the date format of Monolog.